src/main.rs

```diff
@@ -29,41 +29,56 @@ fn force_update_async() -> Result<()> {
 }
 
 // Runs bkt after main() handles flag parsing
-#[allow(clippy::too_many_arguments)]
-fn run(root_dir: Option<PathBuf>, discard_failures: bool, scope: Option<&str>,
-       mut command: CommandDesc, use_cwd: bool, env_keys: BTreeSet<&OsStr>, ttl: Duration,
-       stale: Option<Duration>, warm: bool, force: bool) -> Result<i32> {
-    assert!(!ttl.as_secs() > 0 || ttl.subsec_nanos() > 0, "--ttl cannot be zero"); // TODO use is_zero once stable
+fn run(cli: Cli) -> Result<i32> {
+    let ttl: Duration = cli.ttl.into();
+    let stale: Option<Duration> = cli.stale.map(Into::into);
+
+    assert!(
+        !ttl.as_secs() > 0 || ttl.subsec_nanos() > 0,
+        "--ttl cannot be zero"
+    ); // TODO use is_zero once stable
     if let Some(stale) = stale {
```

Comment:
Can you do `if let Some(stale) = cli.stale.map...` so you don't need the separate `let` above?

Reply:
It's also used further down, so I'd have to duplicate that call...

Reply:
Oh sure, this is fine then.